draft-ietf-quic-transport.md

```diff
@@ -3246,7 +3246,21 @@ Implementors are encouraged to register version numbers of QUIC that they are
 using for private experimentation on the GitHub wiki at
 \<https://github.com/quicwg/base-drafts/wiki/QUIC-Versions\>.
 
-
+## Version Aliases
+
+In order to avoid ossification of the version number defined by this draft,
+servers announce a list of version numbers that they interpret as an alias for
+the version number used in this draft. Alias versions MUST NOT be a reserved
+version. Servers SHOULD send at least one version alias, and SHOULD frequently
+change the value that they announce. Each version alias contains a lifetime,
+which indicates how long the server will accept this version alias. It also
+contains an initial salt, which is used instead of the initial salt as defined
```

Comment:
We should think hard about using a special salt vs. the v1 salt. Marten has, almost incidentally, engineered a weak form of SNI encryption: if you _really_ want to decode the initial, you can connect to the server yourself and likely get the salt. But a typical firewall won't be able to read the SNI anymore. This is a victory for the end-to-end principle, but perhaps a Pyrrhic one if firewalls just give up and drop QUIC.  I don't have a particularly strong opinion on how to proceed, but want the WG to make this decision deliberately.

Reply:
Even without the salt, the presence of multiple QUIC versions with different salts would force a middle-box to opportunistically decrypt initials using different salts. So if we decide QUIC being blocked because of too much obfuscation is a consideration, this proposal adds to this level even without a salt.